api_fhir_r4/views/filters/requestParameterFilter.py

```diff
@@ -0,0 +1,154 @@
+from abc import ABC
+from datetime import timedelta
+
+from api_fhir_r4.utils import TimeUtils
+from core.datetimes.ad_datetime import datetime
+
+
+class QuerysetFilterABC(ABC):
+    def __init__(self, field, value):
+        self.field = field
+        self.value = value
+
+    def apply_filter(self, queryset):
+        """
+        apply_filter should apply contained filter to given queryset and return the result.
+        @param queryset: queryset the filter should be applied to
+        @return: filtered queryset
+        """
+        raise NotImplementedError('apply_filter() not implemented')
+
+
+class QuerysetEqualFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        return queryset.filter(**{self.field: self.value})
+
+
+class QuerysetNotEqualFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        return queryset.exclude(**{self.field: self.value})
+
+
+class QuerysetGreaterThanFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        return queryset.filter(**{f'{self.field}__gt': self.value})
+
+
+class QuerysetLesserThanFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        return queryset.filter(**{f'{self.field}__lt': self.value})
+
+
+class QuerysetGreaterThanEqualFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        return queryset.filter(**{f'{self.field}__gte': self.value})
+
+
+class QuerysetLesserThanEqualFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        return queryset.filter(**{f'{self.field}__lte': self.value})
+
+
+class QuerysetApproximateDateFilter(QuerysetFilterABC):
+    def apply_filter(self, queryset):
+        range_size = (TimeUtils.now() - self.value).days * 0.1
+        value_start = self.value - timedelta(days=range_size)
+        value_end = self.value + timedelta(days=range_size)
+        return queryset.filter(**{f'{self.field}__range': (value_start, value_end)})
+
+
+class QuerysetParameterABC(ABC):
+    def __init__(self, output_parameter):
+        self.output_parameter = output_parameter
+        self.accepted_prefixes = self._get_prefix_filter_mapping().keys()
+
+    def _get_prefix_filter_mapping(self):
+        """
+        _get_prefix_filter_mapping should return a dict that maps respective prefixes from FHIR specification to lambdas
```

Comment:
It's great that you added docsting. However if you're explaining what value given method is returning it'd be recommended to also attach TypeHint. In case of this method it'd be something like: 
``` python
def create(self, validated_data) -> Dict[str, Callable[[str, str], QueryFilterABC]]:
    ...
```